mycity/mycity/intents/get_alerts_intent.py

```diff
@@ -77,27 +77,69 @@ def get_alerts_intent(
     :param get_alerts_function_for_test: Injectable function for unit tests
     :param prune_normal_responses_function_for_test: Injectable function
      for unit tests
-    :param alerts_to_speech_output_function_for_test: Injectable function 
+    :param alerts_to_speech_output_function_for_test: Injectable function
     for unit tests
     :return: MyCityResponseDataModel object
     """
     logger.debug('MyCityRequestDataModel received:' +
                  mycity_request.get_logger_string())
 
-    alerts = get_alerts() if get_alerts_function_for_test is None \
-        else get_alerts_function_for_test()
-    logger.debug("[dictionary with alerts scraped from boston.gov]:\n" +
-                 str(alerts))
+    # get the intent_variables and sessions_attribute object from the request
+    intent_variables = mycity_request.intent_variables
+    session_attributes = mycity_request.session_attributes
+    decision = intent_variables['Decision'].get('value') \
+        if 'Decision' in intent_variables else None
+    session_alerts = session_attributes.get('alerts')
 
-    pruned_alerts = prune_normal_responses(alerts) \
-        if prune_normal_responses_function_for_test is None \
-        else prune_normal_responses_function_for_test(alerts)
-    logger.debug("[dictionary after pruning]:\n" + str(alerts))
+    logger.debug('decision: ' + str(decision) +
+                 ', session_alerts: ' + str(session_alerts))
 
+    # Build the response.
+    #   - if decision was made before asking alert, remind to ask for alert
+    #   - if there is no decision - probably asking for alerts, list alerts
+    #   - if there is decision w/ alerts, get the appropriate alert
```

Comment:
Thank you for adding some comments! I would have these match the order of the branching code, or move each of these statements into the branches. Putting in near the code itself can make it easier for other developers to know that it needs to be updated if code in that branch changes in the future instead of needing to know there are relevant comments further up the function.

May also be useful to add what some indication of your overall approach of saving alerts to session attributes while deciding what alert type the user wants.